url70ed135da7_ralfbiedert_jcores_tgz-pJ8-sandbox_SimpleFilterTestJ8/src/net/jcores/jre/cores/CoreFile.java

```diff
@@ -0,0 +1,626 @@
+                    return null;
+                } catch (IOException e) {
+                    cc.report(MessageType.EXCEPTION, "Error reading data() from " + x + ". IOException!");
+                    return null;
+                }
+            }
+        }).array(ByteBuffer.class));
+    }
+
+    /**
+     * Deletes the given file objects, recursively. Also deletes directories. Unless the
+     * files or directories are write protected or locked they should be gone afterwards.<br/>
+     * <br/>
+     * 
+     * Examples:
+     * <ul>
+     * <li><code>$(".").dir().filter(".*png$").delete()</code> - Deletes all PNG files below the given directory.</li>
+     * </ul>
+     * 
+     * Multi-threaded.<br/>
+     * <br/>
+     * 
+     * @return The same core file object (<code>this</code>).
+     */
+    public CoreFile delete() {
+        final CommonCore cc = this.commonCore;
+        map(new F1<File, Void>() {
+            public Void f(File x) {
+                int lastSize = Integer.MAX_VALUE;
+                List<File> list = CoreKeeper.$(x).dir(ListDirectories.DO).list();
+
+                while (list.size() < lastSize) {
+                    lastSize = list.size();
+
+                    for (File file : list) {
+                        file.delete();
+                    }
+
+                    list = CoreKeeper.$(x).dir(ListDirectories.DO).list();
+                }
+
+                // Try to delete the entry
+                if (!x.delete()) {
+                    cc.report(MessageType.EXCEPTION, "Unable to delete " + x);
+                }
+
+                return null;
+            }
+        });
+
+        return this;
+    }
+
+    /**
+     * De-serializes the previously serialized {@link CoreObject} from the enclosed file.
+     * Objects that are not serializable are ignored.<br/>
+     * <br/>
+     * 
+     * 
+     * Examples:
+     * <ul>
+     * <li><code>$("storage.ser").file().deserialize(String.class)</code> - Restores a {@link Core} that previously
+     * contained a number of Strings and that was written with <code>core.serialize("storage.ser")</code>.</li>
+     * </ul>
+     * 
+     * Single-threaded. Size-of-one.<br/>
+     * <br/>
+     * 
+     * @param <T> Type of the returned core's content.
+     * @param type Type of the returned core's content.
+     * @param options Currently not used.
+     * @return The previously serialized core (using <code>.serialize()</code>).
+     * @see CoreObject
+     */
+    public <T> CoreObject<T> deserialize(Class<T> type, Option... options) {
+
+        if (size() > 1)
+            this.commonCore.report(MessageType.MISUSE, "deserialize() should not be used on cores with more than one class!");
+
+        // Try to restore the core, and don't forget to set the commonCore
+        try {
+            final CoreObject<T> core = Streams.deserializeCore(type, new FileInputStream(get(0)), this.commonCore);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor: *FileInputStream*. Currently, there are execution paths that do not contain closure statements, for example, when Streams.deserializeCore throws an exception. Ensure the object returned by *FileInputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).